staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -613,6 +613,9 @@ func (config *inClusterClientConfig) ClientConfig() (*restclient.Config, error)
 	// in-cluster configs only takes a host, token, or CA file
 	// if any of them were individually provided, overwrite anything else
 	if config.overrides != nil {
+		if insecure := config.overrides.ClusterInfo.InsecureSkipTLSVerify; insecure {
```

Comment:
not familiar with this logic, but the comment above says  `in-cluster configs only takes a host, token, or CA file`

Reply:
are we just enabling this flag, or do we need to allow all of them ...?

Reply:
I see there is some merge logic for combining flags with config in other flows.. should this take a similar approach?

Reply:
if we explicitly decide to not allow overriding in-cluster-config to be insecure, we should at least document that here and add a unit test to solidify that behavior as intentional

Reply:
Hi @deads2k @enj, do you agree @liggitt's idea? If so, I will close this PR and raise another PR to add a new test case as suggested by @liggitt.